addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ParamGuesserScanController.java

```diff
@@ -0,0 +1,254 @@
+            paramGuesserScanLock.unlock();
+        }
+    }
+
+    @Override
+    public void pauseScan(int id) {
+        paramGuesserScanLock.lock();
+        try {
+            if (this.paramGuesserScanMap.containsKey(id)) {
+                this.paramGuesserScanMap.get(id).pauseScan();
+                updateScansPanel();
+            }
+        } finally {
+            paramGuesserScanLock.unlock();
+        }
+    }
+
+    @Override
+    public void pauseAllScans() {
+        paramGuesserScanLock.lock();
+        try {
+            paramGuesserScanMap.values().forEach(GuesserScan::pauseScan);
+            updateScansPanel();
+        } finally {
+            paramGuesserScanLock.unlock();
+        }
+    }
+
+    @Override
+    public void resumeScan(int id) {
+        paramGuesserScanLock.lock();
+        try {
+            if (this.paramGuesserScanMap.containsKey(id)) {
+                this.paramGuesserScanMap.get(id).resumeScan();
+                updateScansPanel();
+            }
+        } finally {
+            paramGuesserScanLock.unlock();
+        }
+    }
+
+    @Override
+    public void resumeAllScans() {
+        paramGuesserScanLock.lock();
+        try {
+            paramGuesserScanMap.values().forEach(GuesserScan::resumeScan);
+            updateScansPanel();
+        } finally {
+            paramGuesserScanLock.unlock();
+        }
+    }
+
+    @Override
+    public GuesserScan removeScan(int id) {
+        paramGuesserScanLock.lock();
+        try {
+            if (!this.paramGuesserScanMap.containsKey(id)) {
+                return null;
+            }
+            GuesserScan currentScan = this.paramGuesserScanMap.get(id);
+            currentScan.stopScan();
+            currentScan.clear();
+            this.paramGuesserScanMap.remove(id);
+            return currentScan;
+        } finally {
+            paramGuesserScanLock.unlock();
+        }
+    }
+
+    @Override
+    public int removeAllScans() {
+        paramGuesserScanLock.lock();
+        try {
+            int count = 0;
+            for (Iterator<GuesserScan> it = paramGuesserScanMap.values().iterator();
+                    it.hasNext(); ) {
+                GuesserScan ascan = it.next();
+                ascan.stopScan();
+                ascan.clear();
+                it.remove();
+                count++;
+            }
+            return count;
+        } finally {
+            paramGuesserScanLock.unlock();
+        }
+    }
+
+    @Override
+    public int removeFinishedScans() {
+        paramGuesserScanLock.lock();
+        try {
+            int count = 0;
+            for (Iterator<GuesserScan> it = paramGuesserScanMap.values().iterator();
+                    it.hasNext(); ) {
+                GuesserScan scan = it.next();
+                if (scan.isStopped()) {
+                    scan.stopScan();
+                    scan.clear();
+                    it.remove();
+                    count++;
+                }
+            }
```

Comment:
Same thing on the iterator.
Use stream/filter in this case?

Reply:
Needs the iterator to remove and it's also counting.